sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/GcpApiSurfaceTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.gcp;
+
+import static org.apache.beam.sdk.util.ApiSurface.Matchers.classesInPackage;
+import static org.apache.beam.sdk.util.ApiSurface.Matchers.containsOnlyClassesMatching;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import com.google.cloud.bigtable.grpc.BigtableInstanceName;
+import com.google.cloud.bigtable.grpc.BigtableTableName;
+import com.google.common.collect.ImmutableSet;
+import java.io.IOException;
+import java.util.Set;
+import org.apache.beam.sdk.util.ApiSurface;
+import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * API surface verification for {@link org.apache.beam.sdk.io.gcp}.
+ */
+@RunWith(JUnit4.class)
+public class GcpApiSurfaceTest {
+
+  @SuppressWarnings("unchecked")
+  private final Set<Matcher<Class<?>>> allowedClasses =
+      ImmutableSet.of(classesInPackage("com.google.api.client.json"),
+                      classesInPackage("com.google.api.client.util"),
+                      classesInPackage("com.google.api.services.bigquery.model"),
+                      classesInPackage("com.google.auth"),
+                      classesInPackage("com.google.bigtable.v2"),
+                      classesInPackage("com.google.cloud.bigtable.config"),
+                      Matchers.<Class<?>>equalTo(BigtableInstanceName.class),
```

Comment:
I think this can also be done via upcast `equalTo((Class<?>) BigtableInstanceName.class)`.

In a funny way, to match the stuff around, I would prefer the fully qualified

```
equalTo((Class<?>) com.google.cloud.bigtable.grpc.BigtableTableName)
```